src/lib/make-welcome-notification.ts

```diff
@@ -0,0 +1,52 @@
+import Cookies from 'js-cookie'
+import { toast, ToastColor } from 'components/toast'
+
+const WELCOME_COOKIE = 'dev-dot-first-welcome-notification'
+
+const DISMISS_COOKIE = 'dev-dot-dismiss-welcome'
+const SESSION_COOKIE = 'dev-dot-welcome-session'
+const AUTO_DISMISS = 15000 // 15 seconds
+const MAX_DATE = new Date('1/31/2023')
+
+function permanentlyDismiss() {
+	// Set dismiss cookie that expires after a year assuming we won't need these notifications in a year after GA
+	Cookies.set(DISMISS_COOKIE, true, {
+		expires: MAX_DATE,
+	})
+	Cookies.remove(WELCOME_COOKIE)
+	Cookies.remove(SESSION_COOKIE)
+}
+
+export function makeWelcomeToast() {
+	// Return if current date is later than the max date
+	if (new Date() > MAX_DATE) {
+		return
+	}
+
+	// Don't show toast if it has been dismissed or session is still active
+	const dismissWelcomeToast = Cookies.get(DISMISS_COOKIE)
+	const inSession = Cookies.get(SESSION_COOKIE)
+	if (dismissWelcomeToast || inSession) {
+		return
+	}
+
+	// If no cookies have been set, set cookie that expires after 3 months from GA
+	const welcomeToastCookie = Cookies.get(WELCOME_COOKIE)
+	if (!welcomeToastCookie) {
+		Cookies.set(WELCOME_COOKIE, true, {
+			expires: MAX_DATE,
+		})
+	}
```

Comment:
🤔 we aren't currently doing anything with this welcomeToastCookie. Do we still need it if we just always want to render one toast until the session cookie is set?

Reply:
Yeah you're right we don't need it if we are adding the date check. Alternatively we can keep it with the proper expiration, remove the date check, and then check if the `welcomeToastCookie` exists before rendering. What do you think?
https://github.com/hashicorp/dev-portal/pull/995/commits/ef580a5838b31ec49a7645c21f190ca0dd1f8dd4#diff-b043befe629def842d164cb756567813f1e0dcacc5ea16d005ae2a99d8b7188dR42

Reply:
I think retaining the date check and just removing this cookie would be reasonable! That limits the amount of cookies we're storing 😄

Reply:
I'm fine with that too since we are still involving 2 other cookies in the function 👍